Example/EnvChanger/AppDelegate.swift

```diff
@@ -9,61 +9,23 @@
 import UIKit
 import EnvChanger
 
-enum Envs: String, EnvironmentRepresentable {
-    case production = "https://production.server.com/"
-    case staging = "https://staging.server.com/"
-    case development = "https://development.server.com"
-    case testing = "https://10.0.1.1/"
-    case edge = "edge.server.com"
-    
-    var environmentTitle: String {
-        return rawValue
-    }
-}
-
-var ACTIVE_ENVIRONMENT = Envs.production.environmentTitle
-
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
-
     var window: UIWindow?
-
-    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+    
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         
-        let envChanger = EnvChangerController(envs: Envs.self) { selectedEnvironment in
-            ACTIVE_ENVIRONMENT = selectedEnvironment.environmentTitle
-            
-            print(ACTIVE_ENVIRONMENT)
+        /// Instantiate and pass handler for environment selection.
+        let envChanger = EnvChangerController(envs: NetworkService.Environments.self,
+                                              window: window) { selectedEnvironment in
+            NetworkService.activeEnvironment = selectedEnvironment.environmentTitle
         }
         
-        ACTIVE_ENVIRONMENT = envChanger.getSavedEnvironment()
+        /// If an environment has been saved, set is as the active environment.
+        if let environment = envChanger.savedEnvironment {
```

Comment:
The completion handler from the `EnvChangerController` creation would be called asynchronously after setting the saved environment.
So do we want it that way or we can move it to the `handler`.

Reply:
Maybe I'm misunderstanding, but I don't think this block is related to the handler. The handler is called after selecting an environment. Meanwhile, the idea behind this block is to set the initial environment to the one saved in UserDefaults (i.e. the last selected environment if there is one), before a selection is even made. Let me know if I'm missing something.

Reply:
It makes sense what you are saying. I don't get the whole picture from only code and understood it in a different way. Don't make any changes to it.